src/vcpkg/binarycaching.cpp

```diff
@@ -914,6 +915,251 @@ namespace
+                auto cacheId = reserve_cache_entry(prefix, abi, cache_size);
+                if (!cacheId) continue;
+
+                std::vector<std::string> headers{
+                    m_token_header,
+                    m_accept_header.to_string(),
+                    "Content-Type: application/octet-stream",
+                    "Content-Range: bytes 0-" + std::to_string(cache_size) + "/*",
+                };
+                auto url = build_write_url(prefix) + "/" + std::to_string(*cacheId.get());
+                if (!put_file(fs, url, {}, headers, tmp_archive_path, "PATCH"))
+                {
+                    continue;
+                }
+
+                Json::Object commit;
+                commit.insert("size", std::to_string(cache_size));
+                auto cmd = command().string_arg(url).string_arg("-d").string_arg(stringify(commit));
+
+                auto res = cmd_execute_and_capture_output(cmd);
+                if (!res.has_value() || res.get()->exit_code) continue;
+                ++upload_count;
+            }
+
+            msg::println(msgUploadedPackagesToVendor,
+                         msg::count = upload_count,
+                         msg::elapsed = timer.elapsed(),
+                         msg::vendor = "GHA");
+        }
+
+        void precheck(View<InstallPlanAction> actions, View<CacheStatus*> cache_status) const override
+        {
+            std::vector<CacheAvailability> actions_availability{actions.size()};
+            for (const auto& prefix : m_read_prefixes)
+            {
+                for (size_t idx = 0; idx < actions.size(); ++idx)
+                {
+                    auto&& action = actions[idx];
+                    const auto& abi = action.package_abi().value_or_exit(VCPKG_LINE_INFO);
+                    if (!cache_status[idx]->should_attempt_precheck(this))
+                    {
+                        continue;
+                    }
+
+                    if (!lookup_cache_entry(prefix, abi).empty())
+                    {
+                        actions_availability[idx] = CacheAvailability::available;
+                        cache_status[idx]->mark_available(this);
+                    }
+                }
+            }
+
+            for (size_t idx = 0; idx < actions.size(); ++idx)
+            {
+                const auto this_cache_status = cache_status[idx];
+                if (this_cache_status && actions_availability[idx] == CacheAvailability::unavailable)
+                {
+                    this_cache_status->mark_unavailable(this);
+                }
+            }
+        }
+
+        static constexpr StringLiteral m_accept_header = "Accept: application/json;api-version=6.0-preview.1";
+        std::string m_token_header;
+
+        std::string m_read_cache_key;
+        std::string m_write_cache_key;
+
+        const VcpkgPaths& paths;
+        std::vector<std::string> m_read_prefixes;
+        std::vector<std::string> m_write_prefixes;
```

Comment:
Is there a reason for these declarations? It doesn't look like these variables are being used for anything.